ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/wizard/v1/wizard/clientlibs/site/js/wizardview.js

```diff
@@ -301,6 +309,65 @@
             if (tabs && lastVisibleIndex >= 0) {
                 this.#navigateAndFocusTab(lastVisibleIndex);
             }
+            this.#hideUnhideNavButtons(this.#_active, this.#getCachedTabs().length);
+        }
+
+        /**
+         * Hides or shows next or and previous navigation buttons
+         *
+         * @private
+         * @param {Number} index The index of the tab to navigate to
+         * @param {Number} total number of tabs
+         */
+        #hideUnhideNavButtons(activeTabIndex, tabsLength) {
+            let nextVisible = this.#findNextVisibleChildIndex(activeTabIndex);
+            let previousVisible = this.#findLastVisibleChildIndex(activeTabIndex);
+
+            if(tabsLength === 0 || this.maxEnabledTab <= this.minEnabledTab) {
+                this.getPreviousButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+                this.getNextButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+            }
+
+            if(activeTabIndex <= this.minEnabledTab || previousVisible === -1) {
+                this.getPreviousButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+            }
+            if(activeTabIndex === this.maxEnabledTab || nextVisible === -1) {
+                this.getNextButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"false");
+            }
+
+            if(tabsLength > 1 && activeTabIndex > this.minEnabledTab && previousVisible > -1) {
+                this.getPreviousButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"true");
+            }
+
+            if(activeTabIndex < this.maxEnabledTab && nextVisible > -1) {
+                this.getNextButtonDiv().setAttribute(Wizard.DATA_ATTRIBUTE_VISIBLE,"true");
+            }
+        }
+
+        setNavigationRange() {
+            let wizardPanels = this.#getCachedWizardPanels();
+            this.maxEnabledTab = wizardPanels.length-1;
+            for (let i = 0; i < wizardPanels.length; i++) {
+                if(!this.childComponentEnabled(this.#getCachedWizardPanels()[i])) {
+                    this.minEnabledTab = i+1;
```

Comment:
how do we know if 0th index is not enabled then 1st index is enabled?
Is the ! sign added. by mistake.

Reply:
it traverses from 1st tab to last and breaks the loop as soon as it finds the visible tab. if the first n tabs are not visible, it will increment the minEnabledTab index